### PR TITLE
Fix lint and test Error introduced by DataLoader2

### PR DESCRIPTION
**Pull request by ejguan**
Per title
- For lint, auto generated by `pre-commit run --all-files`
- For test Error, add `fork` as the start method for multiprocessing. And, skip this test for windows because there is no `fork` on windows system.

### CODE REVIEW

File: test/test_graph.py [hunk 1]
```diff
@@ -164,8 +160,14 @@ def test_reading_service(self) -> None:
...
             self.assertFalse(new_dp.started)
 
+    @unittest.skipIf(IS_WINDOWS, "Fork is required for lambda")
     def test_multiprocessing_reading_service(self) -> None:
         _, (*_, dp) = self._get_datapipes()  # pyre-ignore
+
+        import torch.multiprocessing as mp
+
+        mp.set_start_method("fork")
+
```
Comment (ejguan) [inline test/test_graph.py:170]: Fixing the test failure here.

**Review by NivekT — COMMENTED** [review]
There are some lingering mypy errors.

**Comment by ejguan** [comment]
> There are some lingering mypy errors.

Good catch. Will send a commit to fix mypy.

**Review by VitalyFedyunin — APPROVED** [review]

**Review by NivekT — APPROVED** [review]
LGTM!

**Comment by ejguan** [comment]
Updating due to https://github.com/pytorch/data/pull/396

**Pull request closed without merging**